pkg/controller/storagecluster/storagecluster.go

```diff
@@ -1122,6 +1123,46 @@ func (c *Controller) CreatePodTemplate(
 	}
 	return newTemplate, nil
 }
+func (c *Controller) getCurrentMaxStorageNodesPerZone(
+	zoneMap map[string]uint64,
+	cluster *corev1.StorageCluster,
+	nodeList *v1.NodeList,
+	cloudProvider cloudprovider.Ops,
+) (uint32, error) {
+	storageNodeMap := make(map[string]*storageapi.StorageNode)
+
+	if storagePodsEnabled(cluster) {
+		storageNodeList, err := c.Driver.GetStorageNodes(cluster)
```

Comment:
This doesn't actually return just the storage nodes. It returns all the nodes in the cluster. It is a bit confusing because all the px nodes are called storagenodes from openstorage or sdk perspective.
We will have to check for node pools to see if the nodes actually have storage or not. Refer - https://github.com/libopenstorage/operator/blob/91972a1696924073a6559c46b055e010a4a4419c/drivers/storage/portworx/status.go#L316.